src/engine/server/server.cpp

```diff
@@ -508,6 +510,36 @@ int CServer::MaxClients() const
 	return m_NetServer.MaxClients();
 }
 
+void CServer::InitRconPasswordIfEmpty()
+{
+	if(g_Config.m_SvRconPassword[0])
+	{
+		return;
+	}
+
+	static const char VALUES[] = "ABCDEFGHKLMNPRSTUVWXYZabcdefghjkmnopqt23456789";
+	static const size_t NUM_VALUES = sizeof(VALUES) - 1; // Disregard the '\0'.
+	static const size_t PASSWORD_LENGTH = 6;
+	dbg_assert(NUM_VALUES * NUM_VALUES >= 2048, "need at least 2048 possibilities for 2-character sequences");
+	// With 6 characters, we get a password entropy of log(2048) * 6/2 = 33bit.
+
+	dbg_assert(PASSWORD_LENGTH % 2 == 0, "need an even password length");
+	unsigned short aRandom[PASSWORD_LENGTH / 2];
+	char aRandomPassword[PASSWORD_LENGTH+1];
+	aRandomPassword[PASSWORD_LENGTH] = 0;
+
+	secure_random_fill(aRandom, sizeof(aRandom));
+	for(size_t i = 0; i < PASSWORD_LENGTH / 2; i++)
+	{
+		unsigned short RandomNumber = aRandom[i] % 2048;
+		aRandomPassword[2 * i + 0] = VALUES[RandomNumber / NUM_VALUES];
```

Comment:
The entropy for this characters is slightly lower because `RandomNumber / NUM_VALUES` will never return `45` and the possiblity for `44` is also pretty low. This will generate a lower entropy (~32.6) for the whole password than using a 1-character sequence generated by `aRandom[i] % NumValues` with aRandom as bytes